src/server/rpc/procedures/hurricane-tracking/hurricane-tracking.js

```diff
@@ -0,0 +1,133 @@
+const lineReader = require('line-reader');
+const path = require('path');
+
+const table = [
+    ['name', 'year', 'month', 'day', 'time', 'recordID', 'status', 'latitude', 'longitude', 'maxWind', 'minPressure']
+];
+
+let name = '';
+
+lineReader.eachLine(path.join(__dirname,'hurdat2-1851-2018-051019.txt'), function (line) {
+    if (line.startsWith('AL')){
+        name = line.substring(19, 28).trim();
+    } else {
+        let year = line.substring(0, 4);
+        let month = line.substring(4,6);
+        let day = line.substring(6,8);
+        let time = line.substring(10,14);
+        let recordID = line.substring(15,17);
+        let status = line.substring(18,21).trim();
+        let latitude = line.substring(23,27);
+        let longitude = '-' + line.substring(31,35);
+        let maxWind = line.substring(39,41);
+        let minPressure = line.substring(43,47).trim();
+        let data = [name, year, month, day, time, recordID, status, latitude, longitude, maxWind, minPressure];
+        table.push(data);
+    }
+});
+
+const hurricaneTracker = {};
+
+hurricaneTracker.serviceName = 'HurricaneInfo';
+
+/**
+ * returns a table displaying all of the recorded hurricanes and their respective information
+ * @returns {Array} table - displays hurricane information since 1851
+ */
+
+hurricaneTracker.getFullTable = function(){
+    return table;
+};
+
+/**
+ *
+ * @param {string} name - name of the hurricane
+ * @param {string} year - year that the hurricane occurred in
```

Comment:
It might make more sense to have year be a `BoundedNumber<1850,2020>`. This will make sure the year is a number in the given range rather than a string.

Reply:
(then you won't need to check for valid type for `year` in the actual method)